### PR TITLE
feat: Accept from email intermediate screen

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [X] I ran `npm run re:build`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/login/HSwitchLoginFlow/HyperSwitchEmailVerifyScreen.res [hunk 1]
```diff
@@ -50,7 +42,7 @@ let make = (~setAuthType, ~setAuthStatus) => {
   }
 
   React.useEffect0(() => {
-    let body = generateBody(url)
+    let body = HyperSwitchAuthUtils.generateBody(url)
```
Comment (PritishBudhiraja) [inline src/screens/login/HSwitchLoginFlow/HyperSwitchEmailVerifyScreen.res:45]: generateBody of what type?
add the full name for whatever it is accept invite or any other it is used. If moving to utils else it's used only in this file

File: src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res [hunk 2]
```diff
@@ -457,3 +457,11 @@ let errorSubCodeMapper = (subCode: string) => {
   | _ => UR_00
   }
 }
+
+let generateBody = (url: RescriptReactRouter.url) => {
+  let body = Dict.make()
+  let val = url.search->LogicUtils.getDictFromUrlSearchParams->Dict.get("token")->Option.getOr("")
+
+  body->Dict.set("token", val->JSON.Encode.string)
```
Comment (PritishBudhiraja) [inline src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res:465]:
```Suggestion
let generateBody = (url: RescriptReactRouter.url) => [("token",url.search->LogicUtils.getDictFromUrlSearchParams->Dict.get("token")->Option.getOr("")->JSON.Encode.string)]->LogicUtils.getJsonFromArrayOfJson

File: src/screens/login/HSwitchLoginFlow/HSActivateFromEmailScreen.res [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      let res = await updateDetails(url, body, Post, ())
+      let email = res->JSON.Decode.object->Option.getOr(Dict.make())->getString("email", "")
+      let token = HyperSwitchAuthUtils.parseResponseJson(
+        ~json=res,
+        ~email,
+        ~isAcceptInvite=acceptInvite,
+      )
+
+      if !(token->isEmptyString) && !(email->isEmptyString) {
+        setAuthStatus(LoggedIn(HyperSwitchAuthTypes.getDummyAuthInfoForToken(token)))
```
Comment (PritishBudhiraja) [inline src/screens/login/HSwitchLoginFlow/HSActivateFromEmailScreen.res:24]: Already open
```Suggestion
setAuthStatus(LoggedIn(getDummyAuthInfoForToken(token)))

**Review by PritishBudhiraja — CHANGES_REQUESTED** [review]

File: src/screens/login/HSwitchLoginFlow/HyperSwitchEmailVerifyScreen.res (hunk 1 above)
Comment (Riddhiagrawal001) [reply src/screens/login/HSwitchLoginFlow/HyperSwitchEmailVerifyScreen.res]: Renamed the function to `generateBodyForEmailRedirection` and moved to util file as it is used in both Email verify screen and Accept invite from email screen

File: src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res [hunk 4]
```diff
@@ -457,3 +457,9 @@ let errorSubCodeMapper = (subCode: string) => {
   | _ => UR_00
   }
 }
+
+let generateBodyForEmailRedirection = (url: RescriptReactRouter.url) => {
+  let tokenFromUrl =
+    url.search->LogicUtils.getDictFromUrlSearchParams->Dict.get("token")->Option.getOr("")
+  [("token", tokenFromUrl->JSON.Encode.string)]->LogicUtils.getJsonFromArrayOfJson
```
Comment (PritishBudhiraja) [inline src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res:464]: Can be done in single line. 

We are not using that value and easy code readability.

```Suggestion
[("token",url.search->LogicUtils.getDictFromUrlSearchParams->Dict.get("token")->Option.getOr("")->JSON.Encode.string)]->LogicUtils.getJsonFromArrayOfJson

File: src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res (hunk 2 above)
Comment (Riddhiagrawal001) [reply src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res]: Updated the function

File: src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res (hunk 4 above)
Comment (PritishBudhiraja) [reply src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res]: not needed any separate variable until you are using that somewhere

Comment (PritishBudhiraja) [reply src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res]: Sure. OK

**Review by PritishBudhiraja — DISMISSED** [review]

File: src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res [hunk 5]
```diff
@@ -457,3 +457,10 @@ let errorSubCodeMapper = (subCode: string) => {
   | _ => UR_00
   }
 }
+
+let generateBodyForEmailRedirection = (url: RescriptReactRouter.url) => {
+  open LogicUtils
+
+  let tokenFromUrl = url.search->getDictFromUrlSearchParams->Dict.get("token")->Option.getOr("")
```
Comment (JeevaRamu0104) [inline src/screens/login/HSwitchLoginFlow/HyperSwitchAuthUtils.res:464]: don't set the default value as a empty string

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**